test/InjectorTest.php

```diff
@@ -417,22 +418,6 @@ public function testContainerItselfIsInjectedIfHasReturnsFalse($typeName)
         $this->assertSame($container, $result->result);
     }
 
-    public function testTypeUnavailableInContainerThrowsException()
```

Comment:
The removal of this test is a bit weird: this seems like a common scenario?

Reply:
This was removed for performance and domain responsibility reasons. The psr container should, as per  interface phpdoc, throw when `get()` is not possible. This would in almost every case perform a duplicate `has()` check when an instance is available. Also there are still scenarios when a container returns true for `has()` but `get()` still fails due to additional runtime conditions. IMHO keeping this adds uneccessary complexity.
But I get your point and I should change this to test to ensure Exceptions thrown by the container will not be caught (and therefore hidden). This is the more common scenario.

Reply:
>  IMHO keeping this adds uneccessary complexity.

Changing this is still a BC break

Reply:
True, I already got an idea to fix this. Thanks for your Input.